tiltak/src/main/kotlin/no/nav/amt/tiltak/tiltak/controllers/NavAnsattGjennomforingController.kt

```diff
@@ -0,0 +1,79 @@
+package no.nav.amt.tiltak.tiltak.controllers
+
+import no.nav.amt.tiltak.common.auth.AuthService
+import no.nav.amt.tiltak.common.auth.Issuer
+import no.nav.amt.tiltak.core.port.GjennomforingService
+import no.nav.amt.tiltak.core.port.NavAnsattService
+import no.nav.amt.tiltak.tiltak.repositories.GjennomforingerPaEnheterQuery
+import no.nav.security.token.support.core.api.ProtectedWithClaims
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+import java.time.LocalDate
+import java.util.*
+
+@RestController
+@RequestMapping("/api/nav-ansatt/gjennomforing")
+class NavAnsattGjennomforingController(
+	private val authService: AuthService,
+	private val navAnsattService: NavAnsattService,
+	private val gjennomforingerPaEnheterQuery: GjennomforingerPaEnheterQuery,
+	private val gjennomforingService: GjennomforingService
+) {
+
+	@ProtectedWithClaims(issuer = Issuer.AZURE_AD)
+	@GetMapping
+	fun hentGjennomforinger(): List<HentAlleGjennomforingDto> {
+		val navIdent = authService.hentNavIdentTilInnloggetBruker()
+		val enheter = navAnsattService.hentTiltaksansvarligEnhetTilganger(navIdent)
+
+		return gjennomforingerPaEnheterQuery.query(enheter.map { it.kontor.id })
+			.map { HentAlleGjennomforingDto(it.id, it.navn) }
+	}
+
+	@ProtectedWithClaims(issuer = Issuer.AZURE_AD)
+	@GetMapping("/{gjennomforingId}")
+	fun hentGjennomforing(@PathVariable gjennomforingId: UUID): GjennomforingDto {
+		val navIdent = authService.hentNavIdentTilInnloggetBruker()
+		val enheter = navAnsattService.hentTiltaksansvarligEnhetTilganger(navIdent)
+
+		val gjennomforing = gjennomforingService.getGjennomforing(gjennomforingId)
+
+		if (!enheter.any { it.kontor.id == gjennomforing.navKontorId }) {
+			throw ResponseStatusException(HttpStatus.FORBIDDEN, "Ikke tilgang til enhet")
+		}
+
+		return GjennomforingDto(
```

Comment:
Foretrekker å ha dto i separat fil

Reply:
Personlig så mener jeg det motsatte 😓 
Argumentet mitt for å ha det i samme fil er fordi det aldri skal brukes utenfor controlleren.
Og jeg føler det gir bedre oversikt når det ligger nært hverandre enn å splitte det ut i en egen dto-mappe